lib/host-sp-messages/src/lib.rs

```diff
@@ -175,6 +187,183 @@ pub enum KeyLookupResult {
     MaxResponseLenTooShort,
 }
 
+/// Results for an inventory data request
+///
+/// These **cannot be reordered**; the host and SP must agree on them.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+pub enum InventoryDataResult {
```

Comment:
That's a fair question; it's for ease of serialization / deserialization.

See [RFD 316](https://rfd.shared.oxide.computer/rfd/0316#sp8594hss_commands) for details, but in short, having this be a single `enum` that includes an `Ok` variant makes it constant size, instead of being either `[0]` (for `Ok(())`) or `[1, i]` (for `Err(..)`).